warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/facet/FacetValue.java

```diff
@@ -0,0 +1,85 @@
+package datawave.ingest.mapreduce.handler.facet;
+
+import com.clearspring.analytics.stream.cardinality.HyperLogLogPlus;
+import com.clearspring.analytics.stream.frequency.CountMinSketch;
+import org.apache.accumulo.core.data.Value;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class FacetValue extends Value {
+    private HyperLogLogPlus cardinalityEstimate;
+    private CountMinSketch frequencyEstimate;
+    protected final AtomicBoolean written = new AtomicBoolean(false);
+    
+    public FacetValue(HyperLogLogPlus cardinalityEstimate, CountMinSketch frequencyEstimate) {
+        this.cardinalityEstimate = cardinalityEstimate;
+        this.frequencyEstimate = frequencyEstimate;
+    }
+    
+    public FacetValue(HyperLogLogPlus cardinalityEstimate) {
+        this(cardinalityEstimate, null);
+    }
+    
+    protected FacetValue() {
+        
+    }
+    
+    public static FacetValue buildFrom(final DataInput in) throws IOException {
+        FacetValue value = new FacetValue();
+        value.readFields(in);
+        return value;
+    }
+    
+    public byte[] get() {
+        try {
+            ensureWritten();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return super.get();
+    }
+    
+    @Override
+    public void readFields(DataInput in) throws IOException {
+        int cardSize = in.readInt();
+        byte[] cardinality = new byte[cardSize];
+        in.readFully(cardinality);
+        
+        // TODO: frequencyEstimate
```

Comment:
yup, missing that

Reply:
Strangely, `FacetValue` is never used in this code.  [`FacetHandler.estimate(..)`](https://github.com/drewfarris/datawave/blob/facethandler/warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/facet/FacetHandler.java#L323-L329)  is the only place it's created. Instead of the `FacetValue` a [Simpler Value](https://github.com/drewfarris/datawave/blob/facethandler/warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/facet/FacetHandler.java#L241-L241) is used for the facets. 

Bottom line: I need to do more digging to uncover the original intent here, but I'm hoping that might not prevent merging the current implementation

Reply:
ok